src/azure-cli-core/azure/cli/core/tests/test_generic_update.py

```diff
@@ -39,69 +70,71 @@ def my_set(**kwargs): #pylint:disable=unused-argument
-        self.assertEqual(my_obj['prop'], 'val3', 'set simple property again')
+        app.execute('update-obj --set myDict.myCamelKey=success'.split())
+        self.assertEqual(my_obj.my_dict['myCamelKey'], 'success', 'set simple dict element with camel case key')
 
-        app.execute('gencommand --set list[0]=f'.split())
-        self.assertEqual(my_obj['list'][0], 'f', 'set simple list element')
+        app.execute('update-obj --set myDict.my_snake_key=success'.split())
+        self.assertEqual(my_obj.my_dict['my_snake_key'], 'success', 'set simple dict element with snake case key')
 
-        app.execute('gencommand --set list[2][0]=g'.split())
-        self.assertEqual(my_obj['list'][2][0], 'g', 'set nested list element')
+        # Test the different ways of indexing into a list of objects or dictionaries by filter
+        app.execute('update-obj --set myListOfCamelDicts[myKey=value_2].myKey=new_value'.split())
+        self.assertEqual(my_obj.my_list_of_camel_dicts[1]['myKey'], 'new_value', 'index into list of dictionaries by camel-case key')
 
-        app.execute('gencommand --set list[2][1].d=h'.split())
-        self.assertEqual(my_obj['list'][2][1]['d'], 'h', 'set nested dict element')
+        app.execute('update-obj --set myListOfSnakeDicts[my_key=value2].my_key=new_value'.split())
+        self.assertEqual(my_obj.my_list_of_snake_dicts[1]['my_key'], 'new_value', 'index into list of dictionaries by snake-case key')
 
-        app.execute('gencommand --set list[0]={} list[0].foo=bar'.split())
-        self.assertEqual(my_obj['list'][0]['foo'], 'bar', 'replace nested scalar with new dict')
+        app.execute('update-obj --set myListOfObjects[myString=1.2.3.4].myString=new_value'.split())
+        self.assertEqual(my_obj.my_list_of_objects[1].my_string, 'new_value', 'index into list of objects by key')
 
-        app.execute('gencommand --set list[1]=[] --add list[1] key1=value1'
-                    ' --set list[1][0].key2=value2'.split())
-        self.assertEqual(my_obj['list'][1][0]['key1'], 'value1',
-                         'replace nested scalar with new list with one value')
-        self.assertEqual(my_obj['list'][1][0]['key2'], 'value2',
-                         'add a second value to the new list')
+        # Test setting on elements nested within lists
+        app.execute('update-obj --set myList[1][1]=newValue'.split())
+        self.assertEqual(my_obj.my_list[1][1], 'newValue', 'set nested list element')
 
-        app.execute('gencommand --set list[2][d=f].d=g'.split())
-        self.assertEqual(my_obj['list'][2][2]['d'], 'g',
-                         'index dictionary by unique key=value')
+        app.execute('update-obj --set myList[2].myKey=newValue'.split())
+        self.assertEqual(my_obj.my_list[2]['myKey'], 'newValue', 'set nested dict element')
 
-        app.execute('gencommand --add list i=j k=l'.split())
-        self.assertEqual(my_obj['list'][-1]['k'], 'l',
```

Comment:
In the revision, lines 118-126 do this. It adds a bunch of different things to the list and verifies the order. I replace the final index with [-1]. What do you see that this covers that the new test doesn't?

---
In reply to: [78653857](https://github.com/Azure/azure-cli/pull/867#discussion_r78653857) [](ancestors = 78653857)